### PR TITLE
`libdbmodel`: retry all HTTP requests

### PR DESCRIPTION
**Pull request by lewurm**

### CODE REVIEW

**Comment by lewurm** [comment]
this should help with the failures that we see on the performance bots while interacting with the HTTP API:
```
Unhandled Exception:
System.Threading.Tasks.TaskCanceledException: A task was canceled.
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x7ff5523ff840 + 0x00074> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x7ff5523ff7c0 + 0x0006b> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x7ff5523ff770 + 0x0003a> in <filename unknown>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () <0x7ff5523ffca0 + 0x00012> in <filename unknown>:0 
  at System.IO.Stream+<CopyToAsyncInternal>c__async0.MoveNext () <0x7ff55268bcf0 + 0x002a8> in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x7ff552401860 + 0x00029> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x7ff5523ff840 + 0x0004b> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x7ff5523ff7c0 + 0x0006b> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x7ff5523ff770 + 0x0003a> in <filename unknown>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () <0x7ff5523ffca0 + 0x00012> in <filename unknown>:0 
  at System.Net.Http.HttpContent+<LoadIntoBufferAsync>c__async1.MoveNext () <0x41ee42b0 + 0x00311> in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x7ff552401860 + 0x00029> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x7ff5523ff840 + 0x0004b> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x7ff5523ff7c0 + 0x0006b> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x7ff5523ff770 + 0x0003a> in <filename unknown>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () <0x7ff5523ffca0 + 0x00012> in <filename unknown>:0 
  at System.Net.Http.HttpClient+<SendAsyncWorker>c__async0.MoveNext () <0x41e82fb0 + 0x0078c> in <filename unknown>:0 
[...]
```
source: https://performancebot.mono-project.com/builders/debian-amd64_benchmarker_auto-sgen-noturbo-binary/builds/814/steps/raytracer2/logs/stdio

File: tools/compare/compare.cs [hunk 1]
```diff
@@ -720,7 +720,7 @@ public static int Main (string[] args)
 
 		Console.WriteLine (JsonConvert.SerializeObject (runSet.AsDict ()));
 
-		var uploadResult = AsyncContext.Run (() => Utils.RunWithRetry (() => runSet.Upload ()));
+		var uploadResult = AsyncContext.Run (() => runSet.Upload ());
```
Comment (schani) [inline tools/compare/compare.cs]: Why don't we run this with retry anymore?

Comment (schani) [inline tools/compare/compare.cs]: Oh, because it happens at the request level now!  Nevermind.

**Comment by schani** [comment]
Looks good!

File: tools/compare/compare.cs (hunk 1 above)
Comment (lewurm) [inline tools/compare/compare.cs]: exactly

**Pull request merged**